pageserver/src/feature_resolver.rs

```diff
@@ -0,0 +1,108 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use posthog_client_lite::{FeatureStore, PostHogClient, PostHogEvaluationError};
+use tokio_util::sync::CancellationToken;
+use utils::id::TenantId;
+
+use crate::config::PageServerConf;
+
+struct FeatureResolverInner {
+    posthog_client: PostHogClient,
+    feature_store: Arc<std::sync::Mutex<FeatureStore>>,
```

Comment:
We can eliminate mutex with ArcSwap primitive (here or inside FeatureStore), because we always replace all feature flags.